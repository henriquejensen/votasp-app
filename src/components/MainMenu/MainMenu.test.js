import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import firebase from 'firebase/app';

import MainMenu from './MainMenu';
import HamburgerMenu from '../HamburgerMenu/HamburgerMenu';

describe('MainMenu', () => {
  it('should match a snapshot with no currentUser', () => {
    const history = { push: jest.fn() };
    firebase.auth = jest.fn().mockReturnValue({ currentUser: false });
    expect(toJson(shallow(( <MainMenu.WrappedComponent history={history} /> )))).toMatchSnapshot();
  })

  it('should match a snapshot with currentUser', () => {
    const history = { push: jest.fn() };
    firebase.auth = jest.fn().mockReturnValue({ currentUser: true });
    expect(toJson(shallow(( <MainMenu.WrappedComponent history={history} /> )))).toMatchSnapshot();
  })

  it('should toggle main menu when clicked in HamburgerMenu', () => {
    const history = { push: jest.fn() };
    firebase.auth = jest.fn().mockReturnValue({ currentUser: true });
    const Component = shallow(( <MainMenu.WrappedComponent history={history} /> ));

    expect(Component.state().isOpen).toBeFalsy();
    Component.find(HamburgerMenu).simulate('click');
    expect(Component.state().isOpen).toBeTruthy();
  })

  it('should handle logout correclty', () => {
    const history = { push: jest.fn() };
    firebase.auth = jest.fn().mockReturnValue({ currentUser: true, signOut: jest.fn() });
    const Component = shallow(( <MainMenu.WrappedComponent history={history} /> ));

    Component.find('button').simulate('click');
    expect(history.push).toHaveBeenCalledWith('/app');
  })
})