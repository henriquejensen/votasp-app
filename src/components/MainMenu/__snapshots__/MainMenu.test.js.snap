// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu should match a snapshot with currentUser 1`] = `
<nav
  className="navigation-menu"
>
  <HamburgerMenu
    onClick={[Function]}
  />
  <ul
    className="navigation-menu"
  >
    <li
      className="navigation-menu__list"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="navigation-menu__link"
        to="/app/questionario/1"
      >
        Quest√µes
      </NavLink>
    </li>
    <li
      className="navigation-menu__list"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="navigation-menu__link"
        to="/como-funciona"
      >
        Como funciona
      </NavLink>
    </li>
    <li
      className="navigation-menu__list"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="navigation-menu__link"
        to="/app/ranking"
      >
        Ver meu ranking
      </NavLink>
    </li>
    <li
      className="navigation-menu__list"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="navigation-menu__link"
        to="/app/candidatos-e-candidatas"
      >
        Candidatas e Candidatos
      </NavLink>
    </li>
    <li
      className="navigation-menu__list"
    >
      <button
        className="navigation-menu__link"
        onClick={[Function]}
      >
        Sair
      </button>
    </li>
    <li
      className="navigation-menu__list"
    >
      <a
        className="navigation-menu__link btn-contribua btn btn-primary"
        href="https://www.paypal-brasil.com.br/doe/doecompaypal?qc=&qn=Minha+&qs=25#minha-campinas"
        target="_blank"
      >
        Contribua
      </a>
    </li>
  </ul>
</nav>
`;

exports[`MainMenu should match a snapshot with no currentUser 1`] = `
<nav
  className="navigation-menu"
>
  <HamburgerMenu
    onClick={[Function]}
  />
  <ul
    className="navigation-menu"
  >
    <div>
      <li
        className="navigation-menu__list"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="navigation-menu__link"
          to="/app"
        >
          Quero participar!
        </NavLink>
      </li>
    </div>
  </ul>
</nav>
`;
